RTN-023.tex

```diff
@@ -50,6 +50,372 @@
+of hundreds of quanta, and submits this list submitted to a 
+workload/workflow system (such as HT-Condor or PanDA).
+
+\item Monitors runs which have been submitted  
+	(in the HT-Condor case see Fig. \ref{bpsreport}, currently, could be expanded to PanDA) 
+
+\begin{figure}
+\includegraphics[width=\textwidth]{bpsreport.jpg}]
+\caption{Sample BPS report display showing number of quanta for each task within a give run.
+Credit: M. Gower}
+\label{bpsreport}
+\end{figure}
+
+
+\end{enumerate}
+
+PanDA:
+
+\begin{enumerate}
+
+\item Accepts an ordered list of tasks (from BPS), which specify what
+pipelines to run on which inputs.
+
+\item Distributes tasks to worker nodes, which may be widely distributed.
+
+\item Monitors tasks and associated quanta. See Figs. \ref{pandadash0},\ref{tasks1},\ref{tasks4},\ref{tasks5}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{pandadash0.jpg}]
+\caption{Top level PanDA monitoring dashboard of running and recent jobs
+Credit: S. Padolski}
+\label{pandadash0}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{tasks1.jpg}]
+\caption{Next level task list for first job in list
+Credit: H. Lin}
+\label{tasks1}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{tasks4.jpg}]
+\caption{Next level status for isr task (2nd from bottom in prev list).
+Credit: H. Lin}
+\label{tasks4}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{tasks5.jpg}]
+\caption{Next level status for some of the 603 quanta in this task.
+Credit: H. Lin}
+\label{tasks5}
+\end{figure}
+
+\item Allows viewing of log files for individual tasks. See Fig \ref{log0},\ref{log1}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{log0.jpg}]
+\caption{Getting to the stdout/stderr log for this quantum. 
+Credit: H. Lin}
+\label{log0}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{log1.jpg}]
+\caption{A look at stdout/stderr -- if there was an error, one can view it here to help diagnose an issue.
+Credit: H. Lin}
+\label{log1}
+\end{figure}
+
+\item Allows for retries with more memory for individual tasks that fail.
+
+\item More!
+
+\end{enumerate}
+
+Please see \cite{DMTN-168} for more details on running with BPS and PanDA.
+
+\section{Sample BPS 'run' specification yaml}
+
+A 'run' is defined as a run of a BPS-run-spec.yaml through 'bps submit
+BPS-run-spec.yaml'.
+
+This run-spec yaml file defines:
+
+\begin{enumerate}
+
+\item An ordered list of pipetasks to be run: 
+\begin{verbatim}
+pipelineYaml: 
+"$OBS_LSST_DIR/pipelines/imsim/DRP.yaml#singleFrame,multiVisit"
+\end{verbatim}
+These pipetasks may consist of several steps, and these steps may be defined
+in 'includes' in the DRP.yaml, for instance.
+
+\item A target cluster to run on, and resource requests such as 
+maximum memory (which can be different for different pipelines), number of cpus,
+and maximum wallclock time, and re-try attempt limits.
+
+\item A butler configuration yaml
```

Comment:
Perhaps change to:  The location of the butler repository

Future bps:  This will need improvement if using different repository location at different sites as opposed to single central repo and execution butlers.